vessim/data.py

```diff
@@ -0,0 +1,169 @@
+from datetime import datetime, timedelta
+
+import pandas as pd
+
+
+DatetimeLike = Union[str, datetime]
+PandasObject = Union[pd.Series, pd.DataFrame]
+
+
+Datasets: Dict[str, Dict] = {
+    "solcast2022_germany": {
+        "actual": "solcast2022_germany_actual.csv",
+        "forecast": "solcast2022_germany_forecast.csv",
+        "fill_method": "bfill",
+        "static_forecast": False,
+        "url": "https://raw.githubusercontent.com/dos-group/vessim/solcast_data/datasets/solcast2022_germany.zip",
+    },
+    "solcast2022_global": {
+        "actual": "solcast2022_global_actual.csv",
+        "forecast": "solcast2022_global_forecast.csv",
+        "fill_method": "bfill",
+        "static_forecast": False,
+        "url": "https://raw.githubusercontent.com/dos-group/vessim/solcast_data/datasets/solcast2022_global.zip",
+    },
+}
+
+
+def load_dataset(
+    dataset: Union[str, Dict],
+    data_dir: Path,
+    scale: float = 1.0,
+    start_time: Optional[DatetimeLike] = None,
+    use_forecast: bool = True,
+) -> Tuple[PandasObject, Optional[PandasObject], Literal["ffill", "bfill"]]:
+    """Downloads a dataset from the vessim repository, unpacks it and loads data.
+
+    If all files are already present in the directory, the download is skipped.
+
+    Args:
+        dataset: If a string is provided, the TimeSeriesApi is loaded from one of the
+            vessim datasets. Currently available datasets are:
+                `solcast2022_germany` and `solcast2022_global`
+            Otherwise, it should be a Dictionary containing info about the dataset
+            with following entries:
+                `actual`: Name of the file containing the actual data.
+                `forecast`: Name of the file containing the forecasted data. This is
+                    not needed if use_forecast is set to False.
+                `fill_method`: The fill_method of the TimeSeriesApi. If not specified,
+                    `bfill` is used.
+                `static_forecast`: Bool indicating if the forecast is static. If set
+                    to True, the forecast does not contain a `Request Timestamp`, but
+                    if not specified, the forecast is treated as non-static forecast.
+                    This is not needed if use_forecast is set to False.
+                `url`: String with a URL to a zip-file if data not locally available.
+        data_dir: Absolute path to the directory where the data is/should be located.
+        scale: Multiplies all data points with a value. Defaults to 1.0.
+        start_time: Shifts the data so that it starts at this timestamp if specified.
+            Defaults to None.
+        use_forecast: Bool indicating if forecast should be loaded. Default is true.
+
+    Returns:
+        The dataframe of actual data, the optional dataframe of forecast data and the
+        fill_method to be fed into a TimeSeriesApi.
+
+    Raises:
+        RuntimeError if dataset can not be loaded.
+    """
```

Comment:
please let's not copy paste loads of docs across methods, this is meant to get out of sync once we have to maintain it. Is this method called anywhere else but in `TimeSeriesApi.from_dataset`? Why not put the logic there or make this function private?